corehq/apps/app_manager/models.py

```diff
@@ -792,6 +792,43 @@ def short_comment(self):
         return self.comment if len(self.comment) <= 500 else self.comment[:497] + '...'
 
 
+class CaseLoadReference(DocumentSchema):
+    _allow_dynamic_properties = False
+    path = StringProperty()
+    properties = ListProperty(unicode)
+
+
+class CaseSaveReference(DocumentSchema):
+    _allow_dynamic_properties = False
+    case_type = StringProperty()
+    path = StringProperty()
+    properties = ListProperty(unicode)
+    create = BooleanProperty(default=False)
+    close = BooleanProperty(default=False)
+
+
+class CaseReferences(DocumentSchema):
+    _allow_dynamic_properties = False
+    load = DictProperty()
+    save = DictProperty()
```

Comment:
yeah, agreed. this has already been the case historically because of [this validation](https://github.com/dimagi/commcare-hq/blob/master/corehq/apps/app_manager/views/forms.py#L758). I think [this set of tests](https://github.com/dimagi/commcare-hq/pull/14710/files#diff-c1311f89ab8e0af93474ab88b4019498R110) is the closest proxy to an integration test we have, but there definitely isn't an easy way to check if vellum starts changing things up.